src/session/sidebar/search/item_row.rs

```diff
@@ -0,0 +1,141 @@
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use gtk::{glib, CompositeTemplate};
+
+use crate::tdlib::{Chat, User};
+
+mod imp {
+    use super::*;
+    use once_cell::sync::Lazy;
+    use std::cell::RefCell;
+
+    use crate::session::components::Avatar;
+
+    #[derive(Debug, Default, CompositeTemplate)]
+    #[template(string = r#"
+    <interface>
+      <template class="SidebarSearchItemRow" parent="GtkWidget">
+        <child>
+          <object class="ComponentsAvatar" id="avatar">
+            <property name="size">32</property>
+            <binding name="item">
+              <lookup name="item">SidebarSearchItemRow</lookup>
+            </binding>
+          </object>
+        </child>
+        <child>
+          <object class="GtkLabel" id="label">
+            <property name="ellipsize">end</property>
+          </object>
+        </child>
+      </template>
+    </interface>
+    "#)]
+    pub(crate) struct ItemRow {
+        /// A `Chat` or `User`
+        pub(super) item: RefCell<Option<glib::Object>>,
+        #[template_child]
+        pub(super) avatar: TemplateChild<Avatar>,
+        #[template_child]
+        pub(super) label: TemplateChild<gtk::Label>,
+    }
+
+    #[glib::object_subclass]
+    impl ObjectSubclass for ItemRow {
+        const NAME: &'static str = "SidebarSearchItemRow";
+        type Type = super::ItemRow;
+        type ParentType = gtk::Widget;
+
+        fn class_init(klass: &mut Self::Class) {
+            Self::bind_template(klass);
+
+            klass.set_css_name("sidebarsearchitemrow");
+            klass.set_layout_manager_type::<gtk::BoxLayout>();
+        }
+
+        fn instance_init(obj: &glib::subclass::InitializingObject<Self>) {
+            obj.init_template();
+        }
+    }
+
+    impl ObjectImpl for ItemRow {
+        fn properties() -> &'static [glib::ParamSpec] {
+            static PROPERTIES: Lazy<Vec<glib::ParamSpec>> = Lazy::new(|| {
+                vec![glib::ParamSpecObject::new(
+                    "item",
+                    "Item",
+                    "The item of this row",
+                    glib::Object::static_type(),
+                    glib::ParamFlags::READWRITE | glib::ParamFlags::EXPLICIT_NOTIFY,
```

Comment:
We don't really need `CONSTRUCT` in this case since we don't necessary need the property in `constructed` phase. So, in this case it doesn't change anything adding that because we do things in the setter directly.

Reply:
You're correct. I would have seen the sense also rather in the fact that the parameter is better documented.